### ISSUE TITLE
GrpcHttp2HeadersDecoder uses netty internal classes and fails with netty 4.1.10

### ISSUE THREAD

**Issue by denisgaebler** [grpc/grpc-java#2939]
### Expected behavior
Due to an endianess bug in netty I have to use the fixed version netty 4.1.10
Unfortunatly GrpcHttp2HeadersDecoder references:
import io.netty.handler.codec.http2.internal.hpack.Decoder;
which is gone with netty 4.1.10

### Actual behavior
Execution fails with:
Exception in thread "grpc-default-executor-1" java.lang.NoClassDefFoundError: io/netty/handler/codec/http2/internal/hpack/Decoder
	at io.grpc.netty.GrpcHttp2HeadersDecoder.<init>(GrpcHttp2HeadersDecoder.java:85)
	at io.grpc.netty.GrpcHttp2HeadersDecoder$GrpcHttp2ClientHeadersDecoder.<init>(GrpcHttp2HeadersDecoder.java:147)
	at io.grpc.netty.NettyClientHandler.newHandler(NettyClientHandler.java:120)
	at io.grpc.netty.NettyClientTransport.newHandler(NettyClientTransport.java:303)
	at io.grpc.netty.NettyClientTransport.start(NettyClientTransport.java:173)
	at io.grpc.internal.ForwardingConnectionClientTransport.start(ForwardingConnectionClientTransport.java:44)
	at io.grpc.internal.TransportSet.startNewTransport(TransportSet.java:233)
	at io.grpc.internal.TransportSet.obtainActiveTransport(TransportSet.java:203)
	at io.grpc.internal.ManagedChannelImpl$3.getTransport(ManagedChannelImpl.java:739)
	at io.grpc.internal.ManagedChannelImpl$3.getTransport(ManagedChannelImpl.java:677)
	at io.grpc.PickFirstBalancerFactory$PickFirstBalancer$1.get(PickFirstBalancerFactory.java:129)
	at io.grpc.internal.DelayedClientTransport$2.run(DelayedClientTransport.java:271)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

### Steps to reproduce
Use netty-all-4.1.10.Final-SNAPSHOT.jar

### Netty version
4.1.10

### JVM version (e.g. `java -version`)
1.8

### OS version (e.g. `uname -a`)
Windows 10

**Comment by denisgaebler** [comment]
Basically 

import io.netty.handler.codec.http2.internal.hpack.Decoder;

should be now:
import io.netty.handler.codec.http2.HpackDecoder;

Thanks.

**Comment by carl-mastrangelo** [comment]
This is not an issue.  See https://github.com/netty/netty/issues/6579

**Comment by denisgaebler** [comment]
Well, it is an issue, because the Hyperledger uses grpc-java and it in turn uses netty.
Netty has fixed an ENDIANESS issue in 4.1.10 only, which I need for Hyperledger to work between BIGENDIAN and LITTLEENDIAN platforms. Since Netty 4.1.10 provides APIs that do not work with grpc-java, I am stuck. Catch22. Seams impossible to backport the netty changes to 4.1.8 and I also gave up trying to change the grpc -java code to make it work with netty 4.1.10.

**Comment by carl-mastrangelo** [comment]
So, we need netty 4.1.0 to be formally released before we can update our APIs to use it.  Sounds like the best option you have to apply patches to netty locally, rebuild, and deploy with that for the time being.

**Comment by denisgaebler** [comment]
Yep looks like I have to wait.

**Comment by ejona86** [comment]
Because grpc-java is using the unstable HTTP/2 API, you should only use grpc-java with the Netty version we've version-pinned in our POM (currently 4.1.8). Netty 4.1.10 should have what we need, and we'll try to upgrade to it like normal. We work closely with the Netty team to stay on recent versions, but it can take time. Netty 4.1.10 isn't even released yet.

**Comment by denisgaebler** [comment]
The netty team has asked me to test their changes in the original configuration where the error occurred. So this is not possible as long as grpc-java does not exploit the 4.1.10 changes and I have reported it back to them.
Thanks.

**Comment by carl-mastrangelo** [comment]
Close this as a duplicate of https://github.com/grpc/grpc-java/issues/2841
